xml_converter/tests/attribute_testing.py

```diff
@@ -0,0 +1,165 @@
+from dataclasses import dataclass
+import argparse
+import difflib
+import json
+import subprocess
+import re
+import os
+from typing import List, Optional
+
+# Path to compiled C++ executable
+xml_converter: str = "../build/xml_converter"
+json_file_path: str = "../test_cases/test_expected_outputs.json"
+
+
+@dataclass
+class XMLConverter_arguments:
+    arg_input_xml: str = "--input-taco-path"
+    arg_output_xml: str = "--output-taco-path"
+    arg_input_proto: str = "--input-waypoint-path"
+    arg_output_proto: str = "--output-waypoint-path"
+    arg_split_proto: str = "--output-split-waypoint-path"
+
+
+def run_xml_converter(
+    input_xml: Optional[List[str]] = None,
+    output_xml: Optional[List[str]] = None,
+    input_proto: Optional[List[str]] = None,
+    output_proto: Optional[List[str]] = None,
+    split_output_proto: Optional[str] = None,
+) -> subprocess.CompletedProcess[str]:
+
+    # Build the command to execute the C++ program with the desired function and arguments
+    cmd: List[str] = [xml_converter]
+
+    if input_xml:
+        cmd += [XMLConverter_arguments.arg_input_xml] + input_xml
+    if output_xml:
+        cmd += [XMLConverter_arguments.arg_output_xml] + output_xml
+    if input_proto:
+        cmd += [XMLConverter_arguments.arg_input_proto] + input_proto
+    if output_proto:
+        cmd += [XMLConverter_arguments.arg_output_proto] + output_proto
+    if split_output_proto:
+        cmd += [XMLConverter_arguments.arg_split_proto] + [split_output_proto]
+
+    # Run the C++ program and capture its output
+    result = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+
+    return result
+
+
+def compare_files(file_path1: str, file_path2: str) -> List[str]:
+    try:
+        with open(file_path1, 'r') as file1, open(file_path2, 'r') as file2:
+            content1 = file1.readlines()
+            content2 = file2.readlines()
+
+        differ = difflib.Differ()
+        diff = list(differ.compare(content1, content2))
+        filtered_diff: List[str] = []
+        for line in diff:
+            if line.startswith("+ ") or line.startswith("- "):
+                filtered_diff.append(line)
+
+        return filtered_diff
+
+    except (FileNotFoundError, PermissionError) as e:
```

Comment:
PermissionError should probably still throw a error.
FileNotFoundError should probably return a complete file diff of the other, depending on which file was not found. And maybe throw an error still if both are not found.
If you even need to catch errors here at all.